src/backend/distributed/utils/resource_lock.c

```diff
@@ -1060,3 +1061,143 @@ CitusLockTableAclCheck(Oid relationId, LOCKMODE lockmode, Oid userId)
 
 	return aclResult;
 }
+
+
+/*
+ * EnsureCanAcquireLock checks if currect user has the permissions
+ * to acquire a lock on the table and throws an error if the user does
+ * not have the permissions
+ */
+static void
+EnsureCanAcquireLock(Oid relationId, LOCKMODE lockMode)
+{
+	AclResult aclResult = CitusLockTableAclCheck(relationId, lockMode,
+												 GetUserId());
+	if (aclResult != ACLCHECK_OK)
+	{
+		aclcheck_error(aclResult,
+					   get_relkind_objtype(get_rel_relkind(relationId)),
+					   get_rel_name(relationId));
+	}
+}
+
+
+/*
+ * AcquireDistributedLockOnRelations acquire a distributed lock on worker nodes
+ * for given list of relations ids. Worker node list is sorted so that the lock
+ * is acquired in the same order regardless of which node it was run on. Notice
+ * that no lock is acquired on coordinator node if the coordinator is not added
+ * to the metadata.
+ */
+void
+AcquireDistributedLockOnRelations(List *relationIdList, LOCKMODE lockMode)
+{
+	if (!EnableMetadataSync)
+	{
+		return;
+	}
+
+	const char *lockModeText = LockModeToLockModeText(lockMode);
+
+	UseCoordinatedTransaction();
+
+	StringInfo lockRelationsCommand = makeStringInfo();
+	appendStringInfo(lockRelationsCommand, "%s;\n", DISABLE_DDL_PROPAGATION);
+
+	List *relationIdsToLock = NIL;
+	bool partialLockStmt = false;
+
+	Oid relationId = InvalidOid;
+	foreach_oid(relationId, relationIdList)
+	{
+		EnsureCanAcquireLock(relationId, lockMode);
+
+		CitusTableCacheEntry *tableCacheEntry = GetCitusTableCacheEntry(relationId);
+
+		/* skip citus tables which are not seen by workers */
```

Comment:
this check should be done with `ShouldSyncTableMetadata()`. We centralize the decision on whether to sync any metadata or not for a particular table via ShouldSyncTableMetadata.

Any reason we are not using it here?

Reply:
That check was failing to do what I needed it to do here (skip tables distributed by append and range since apparently even though the workers have shards, they cannot see the actual table).

Reply:
Will try changing the `ShouldSyncTableMetadata()` function and see if anything breaks.